process/economics/economicsData.go

```diff
@@ -1,41 +1,113 @@
+	if notGreaterThanZero < 0 {
+		return nil, process.ErrInvalidRewardsValue
+	}
+
+	err = checkValues(economics)
+	if err != nil {
+		return nil, err
+	}
+
 	return &EconomicsData{
-		rewardsValue:        economics.RewardsSettings.RewardsValue,
+		rewardsValue:        rewardsValue,
 		communityPercentage: economics.RewardsSettings.CommunityPercentage,
 		leaderPercentage:    economics.RewardsSettings.LeaderPercentage,
 		burnPercentage:      economics.RewardsSettings.BurnPercentage,
-		minGasPrice:         economics.FeeSettings.MinGasPrice,
-		minGasLimitForTx:    economics.FeeSettings.MinGasLimitForTx,
-		minTxFee:            economics.FeeSettings.MinTxFee,
+		minGasPrice:         minGasPrice,
+		minGasLimitForTx:    minGasLimitForTx,
 		communityAddress:    economics.EconomicsAddresses.CommunityAddress,
 		burnAddress:         economics.EconomicsAddresses.BurnAddress,
+	}, nil
+}
+
+func convertValues(economics *config.ConfigEconomics) (*big.Int, uint64, uint64, error) {
+	conversionBase := 10
+	bitConversionSize := 64
+
+	rewardsValue := new(big.Int)
+	rewardsValue, ok := rewardsValue.SetString(economics.RewardsSettings.RewardsValue, conversionBase)
+	if !ok {
+		return nil, 0, 0, process.ErrInvalidRewardsValue
 	}
+	minGasPrice, err := strconv.ParseUint(economics.FeeSettings.MinGasPrice, conversionBase, bitConversionSize)
+	if err != nil {
+		return nil, 0, 0, process.ErrInvalidMinimumGasPrice
+	}
+	minGasLimitForTx, err := strconv.ParseUint(economics.FeeSettings.MinGasLimitForTx, conversionBase, bitConversionSize)
+	if err != nil {
+		return nil, 0, 0, process.ErrInvalidMinimumGasLimitForTx
+	}
+
+	return rewardsValue, minGasPrice, minGasLimitForTx, nil
+}
+
+func checkValues(economics *config.ConfigEconomics) error {
+	bigBurnPercentage := big.NewFloat(economics.RewardsSettings.BurnPercentage)
+	bigCommunityPercentage := big.NewFloat(economics.RewardsSettings.CommunityPercentage)
+	bigLeaderPercentage := big.NewFloat(economics.RewardsSettings.LeaderPercentage)
+
+	notGreaterOrEqualWithZero := bigBurnPercentage.Cmp(big.NewFloat(0.0))
+	notLessThanOne := big.NewFloat(1.0).Cmp(bigBurnPercentage)
+	if notGreaterOrEqualWithZero < 0 || notLessThanOne < 0 {
+		return process.ErrInvalidRewardsPercentages
+	}
+
+	notGreaterOrEqualWithZero = bigCommunityPercentage.Cmp(big.NewFloat(0.0))
+	notLessThanOne = big.NewFloat(1.0).Cmp(bigCommunityPercentage)
+	if notGreaterOrEqualWithZero < 0 || notLessThanOne < 0 {
+		return process.ErrInvalidRewardsPercentages
+	}
+
+	notGreaterOrEqualWithZero = bigLeaderPercentage.Cmp(big.NewFloat(0.0))
+	notLessThanOne = big.NewFloat(1.0).Cmp(bigLeaderPercentage)
+	if notGreaterOrEqualWithZero < 0 || notLessThanOne < 0 {
+		return process.ErrInvalidRewardsPercentages
+	}
+
+	sumPercentage := new(big.Float)
+	sumPercentage = sumPercentage.Add(bigBurnPercentage, bigCommunityPercentage)
+	sumPercentage = sumPercentage.Add(sumPercentage, bigLeaderPercentage)
+
+	equalsWithOne := sumPercentage.Cmp(big.NewFloat(1.0))
```

Comment:
might change to:
```GO
isNotEqualToOne = sumPercentage.Cmp(big.NewFloat(1.0)) != 0
if isNotEqualToOne{
    //errors
}

```

Reply:
Done.

Reply:
Comparing floats should be done with verification to an epsilon. like fAbs(expectedValue-actualValue) < 0.000001

Reply:
Agree, should have an epsilon for this since big.Float is not implemented as Decimal

Reply:
Done.